table/block_fetcher.cc

```diff
@@ -178,14 +189,21 @@ inline void BlockFetcher::GetBlockContents() {
     // page can be either uncompressed or compressed, the buffer either stack
     // or heap provided. Refer to https://github.com/facebook/rocksdb/pull/4096
     if (got_from_prefetch_buffer_ || used_buf_ == &stack_buf_[0]) {
-      CopyBufferToHeap();
+      CopyBufferToHeapBuf();
```

Comment:
It might be a good idea to add some comments either at the top of this function or in the header about the various buffers and how data moves between them. Its quite confusing otherwise.

Reply:
left a comment for this method